pkg/controllers/routesync/reconcile.go

```diff
@@ -0,0 +1,220 @@
+	// Walk through each local host route and identify any that need to be removed.
+	for i, hostRoute := range currentHostRoutes {
+		if hostRoute.Labels[routeLabel] != r.Namespace {
+			// ignore routes that this controller isn't responsible for
+			continue
+		}
+		if hostRouteNeedsDeletion(allProposedHostRoutes, hostRoute) {
+			routesToDelete = append(routesToDelete, currentHostRoutes[i])
+		}
+	}
+
+	for _, route := range routesToCreate {
+		r.Log.Info("Will create route", "route", route.Name)
+		_, err := r.HostClient.RouteV1().Routes(r.Namespace).Create(route)
+		if err != nil {
+			r.Log.Error(err, "failed to create route")
+			return ctrl.Result{}, err
+		}
+	}
+	for _, route := range routesToUpdate {
+		r.Log.Info("Will update route", "route", route.Name)
+		_, err := r.HostClient.RouteV1().Routes(r.Namespace).Update(route)
+		if err != nil {
+			r.Log.Error(err, "failed to update route")
+			return ctrl.Result{}, err
+		}
+	}
+	for _, route := range routesToDelete {
+		r.Log.Info("Will delete route", "route", route.Name)
+		err := r.HostClient.RouteV1().Routes(r.Namespace).Delete(route.Name, &metav1.DeleteOptions{})
+		if err != nil && !errors.IsNotFound(err) {
+			r.Log.Error(err, "failed to delete route")
+			return ctrl.Result{}, err
+		}
+	}
+
+	return ctrl.Result{}, nil
+}
+
+func hostRouteNeedsDeletion(proposedHostRoutes []*routev1.Route, hostRoute *routev1.Route) bool {
+	for _, route := range proposedHostRoutes {
+		if route.Name == hostRoute.Name {
+			return false
+		}
+	}
+	return true
+}
+
+func (r *RouteSyncReconciler) createSyncRouteFromTarget(currentRoutes []*routev1.Route, tRoute *routev1.Route) (*routev1.Route, processRoute) {
+	targetRouteNamespacedName := fmt.Sprintf("%s-%s", tRoute.Namespace, tRoute.Name)
+	syncedRouteName := r.generateRouteName(targetRouteNamespacedName)
+
+	var existingRoute *routev1.Route
+
+	for i, cRoute := range currentRoutes {
+		if cRoute.Name == syncedRouteName {
+			existingRoute = currentRoutes[i].DeepCopy()
+			break
+		}
+	}
+
+	sRoute := &routev1.Route{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      syncedRouteName,
+			Namespace: r.Namespace,
+			Labels: map[string]string{
+				routeLabel: r.Namespace,
+			},
+		},
+	}
+
+	sRoute.Spec = routev1.RouteSpec{
+		Host: tRoute.Spec.Host,
+	}
+
+	if tRoute.Spec.TLS == nil {
+		sRoute.Spec.To = httpRouteTarget
+	} else {
+		sRoute.Spec.To = httpsRouteTarget
+		sRoute.Spec.TLS = &routev1.TLSConfig{
+			// Always pass through to the target cluster's route and allow the target
+			// cluster to terminate as if traffic had been directed to it.
+			Termination: routev1.TLSTerminationPassthrough,
+			// Use the same edge termination policy as the target route
+			InsecureEdgeTerminationPolicy: tRoute.Spec.TLS.InsecureEdgeTerminationPolicy,
+		}
+	}
+
+	if existingRoute == nil {
+		// No existing route, so we need to create a new one.
+		return sRoute, createRoute
+	}
+
+	if reflect.DeepEqual(existingRoute.Spec, sRoute.Spec) {
```

Comment:
This comparison is failing because the existing route also includes `wildcardPolicy: None` as in:
```
spec:
  host: blahblah
  tls:
    insecureEdgeTerminationPolicy: Redirect
    termination: passthrough
  to:
    kind: Service
    name: router-https
    weight: 100
  wildcardPolicy: None
```
It may be better to compare the fields we're setting individually